samplesheets/views_api.py

```diff
@@ -147,6 +165,256 @@ def post(self, request, *args, **kwargs):
+                    )
+            except Exception as ex:
+                raise ValidationError(
+                    '{} {}'.format('Creating ' + IRODS_TICKET_EX_MSG + ':', ex)
+                )
+
+            # Create database object
+            serializer.validated_data['ticket'] = ticket.ticket
+            irods_access_ticket = serializer.save()
+
+            # Create timeline event and app alerts
+            self.create_timeline_event(irods_access_ticket, 'create')
+            self.create_app_alerts(
+                irods_access_ticket, 'create', self.request.user
+            )
+            context = {
+                'ticket': serializer.data,
+                'detail': IRODS_TICKET_CREATED_MSG,
+            }
+        else:
+            raise ValidationError('Invalid data: {}'.format(serializer.errors))
+        return Response(context, status=status.HTTP_201_CREATED)
+
+
+class IrodsAccessTicketUpdateAPIView(
+    IrodsAccessTicketModifyMixin, SODARAPIBaseProjectMixin, UpdateAPIView
+):
+    """
+    Update an iRODS access ticket for a project.
+
+    **URL:** ``/samplesheets/api/irods/ticket/update/{IrodsAccessTicket.sodar_uuid}``
+
+    **Methods:** ``PUT``, ``PATCH``
+
+    **Parameters:**
+
+    - ``label``: Label
+    - ``date_expires``: Expiration date (YYYY-MM-DD)
+
+    **Returns:**
+
+    - ``ticket``: iRODS access ticket
+    """
+
+    http_method_names = ['put', 'patch']
+    permission_required = 'samplesheets.edit_sheet'
+
+    def get_object(self):
+        return IrodsAccessTicket.objects.get(
+            sodar_uuid=self.kwargs.get('irodsaccessticket')
+        )
+
+    def put(self, request, *args, **kwargs):
+        """PUT request for updating an iRODS access ticket"""
+        ticket = self.get_object()
+        serializer = IrodsAccessTicketSerializer(
+            ticket, data=request.data, partial=True
+        )
+        read_only_fields = set(request.data) & set(
+            serializer.get_read_only_fields()
+        )
+        if read_only_fields:
+            raise ValidationError(
+                'Updating {}: {}: {}'.format(
+                    IRODS_TICKET_EX_MSG,
+                    IRODS_TICKET_READ_ONLY_FIELDS_MSG,
+                    ', '.join(read_only_fields),
+                )
+            )
+
+        if serializer.is_valid():
+            serializer.save()
+        else:
+            raise ValidationError(
+                'Updating {}: {}: {}'.format(
+                    IRODS_TICKET_EX_MSG,
+                    IRODS_TICKET_INVALID_DATA_MSG,
+                    serializer.errors,
+                )
+            )
+        self.create_timeline_event(ticket, 'update')
+        self.create_app_alerts(ticket, 'update', self.request.user)
+        context = {
+            'ticket': serializer.data,
+            'detail': IRODS_TICKET_UPDATED_MSG,
+        }
+        return Response(context, status=status.HTTP_200_OK)
+
+
+class IrodsAccessTicketDeleteAPIView(
```

Comment:
API deletion views should be named `*DestroyAPIView` to be consistent with DRF conventions. (It's still OK to use the word "delete" in API documentation docstrings as the user shouldn't care about this distinction)